### PR TITLE
readme: add info about branches

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/6882.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,21 @@
...
+      to v5.5.4.
+- `legacy`
+    - From version v5.5.4 (inclusive) up to v6.0.0 (exclusive).
+    - Contains only the latest version of legacy controllers (reconciling
+      AWSConfig CRs).
+- `master`
+    - From version v6.0.0 (inclusive).
+    - Contains only the latest version of controllers (reconciling cluster API
+      objects).
+
```
Comment (kopiczko) [inline README.md]: Can someone from @giantswarm/team-ludacris confirm I got the versions right?

File: README.md [hunk 2]
```diff
@@ -4,6 +4,21 @@
...
 The aws-operator manages Kubernetes clusters running on AWS.
 
+## Branches
+
+- `thiccc`
+    - Up to version v5.5.4 (exclusive).
+    - Contains all versions legacy controllers (reconciling AWSConfig CRs) up
+      to v5.5.4.
+- `legacy`
+    - From version v5.5.4 (inclusive) up to v6.0.0 (exclusive).
```
Comment (xh3b4sd) [inline README.md]: Latest in `legacy` is `5.5.0`, aka `v31`.

File: README.md [hunk 3]
```diff
@@ -4,6 +4,21 @@
 
 The aws-operator manages Kubernetes clusters running on AWS.
 
+## Branches
+
+- `thiccc`
+    - Up to version v5.5.4 (exclusive).
+    - Contains all versions legacy controllers (reconciling AWSConfig CRs) up
```
Comment (xh3b4sd) [inline README.md]:
```suggestion
    - Contains all versions of legacy controllers (reconciling AWSConfig CRs) up
```

File: README.md [hunk 4]
```diff
@@ -4,6 +4,21 @@
 
 The aws-operator manages Kubernetes clusters running on AWS.
 
+## Branches
+
+- `thiccc`
+    - Up to version v5.5.4 (exclusive).
```
Comment (kopiczko) [inline README.md]:
```suggestion
    - Up to version v5.4.0.
```

File: README.md (hunk 2 above)
Comment (kopiczko) [inline README.md]:
```suggestion
    - From version v5.5.4 up to v5.x.x.
```

File: README.md [hunk 5]
```diff
@@ -4,6 +4,21 @@
...
+    - Contains all versions legacy controllers (reconciling AWSConfig CRs) up
+      to v5.5.4.
+- `legacy`
+    - From version v5.5.4 (inclusive) up to v6.0.0 (exclusive).
+    - Contains only the latest version of legacy controllers (reconciling
+      AWSConfig CRs).
+- `master`
+    - From version v6.0.0 (inclusive).
```
Comment (kopiczko) [inline README.md]:
```suggestion
    - From version v6.0.0.
```

File: README.md [hunk 6]
```diff
@@ -4,6 +4,21 @@
...
 The aws-operator manages Kubernetes clusters running on AWS.
 
+## Branches
+
+- `thiccc`
+    - Up to version v5.4.0.
+    - Contains all versions of legacy controllers (reconciling AWSConfig CRs) up
+      to v5.5.4.
+- `legacy`
+    - From version v5.5.4 up to v5.x.x.
```
Comment (TheoBrigitte) [inline README.md]:
```suggestion
    - From version v5.5.0 up to v5.x.x.
```

File: README.md [hunk 7]
```diff
@@ -4,6 +4,21 @@
 
 The aws-operator manages Kubernetes clusters running on AWS.
 
+## Branches
+
+- `thiccc`
+    - Up to version v5.4.0.
```
Comment (JosephSalisbury) [inline README.md]: Up to an including, or up to and excluding?

Comment (JosephSalisbury) [reply README.md]: And same further, there is the possibility for ambiguity here.

Comment (kopiczko) [reply README.md]: Everything is inclusive here.

Comment (JosephSalisbury) [reply README.md]: Can we be explicit about that?

Comment (kopiczko) [reply README.md]: Suggestions please <3

Comment (JosephSalisbury) [inline README.md]:
```suggestion
    - Up to and including version v5.4.0.
```

File: README.md [hunk 8]
```diff
@@ -4,6 +4,21 @@
...
+    - Contains all versions of legacy controllers (reconciling AWSConfig CRs) up
+      to v5.4.0.
```
Comment (JosephSalisbury) [inline README.md]:
```suggestion
      to and including v5.4.0.
```

File: README.md [hunk 9]
```diff
@@ -4,6 +4,21 @@
...
+- `legacy`
+    - From version v5.5.0 up to v5.x.x.
```
Comment (JosephSalisbury) [inline README.md]:
```suggestion
    - From version v5.5.0 up to and including v5.x.x.
```

**Comment by kopiczko** [comment]
Is it good now?

**Pull request merged**